driver/cri.go

```diff
@@ -0,0 +1,338 @@
+	}
+
+	driver := &CRIDriver{
+		criSocketAddress: path,
+		runtimeClient:    &runtimeClient,
+		imageClient:      &imageClient,
+		context:          context.Background(),
+		cconfig:          cconfig,
+		pconfig:          pconfig,
+	}
+
+	return driver, nil
+
+}
+
+func getGRPCConn(socket string, timeout time.Duration) (*grpc.ClientConn, error) {
+	conn, err := grpc.Dial(socket, grpc.WithInsecure(), grpc.WithTimeout(timeout),
+		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}))
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect: %v", err)
+	}
+	return conn, nil
+}
+
+// Name returns the name of the container
+func (ctr *CRIContainer) Name() string {
+	return ctr.name
+}
+
+// Detached returns whether the container is to be started in detached state
+func (ctr *CRIContainer) Detached() bool {
+	return true
+}
+
+// Trace returns whether the container should be traced (using any tracing supported
+// by the container runtime)
+func (ctr *CRIContainer) Trace() bool {
+	return ctr.trace
+}
+
+// Image returns either a bundle path (used by runc, containerd) or image name (used by Docker)
+// that will be used by the container runtime to know what image to run/execute
+func (ctr *CRIContainer) Image() string {
+	return ctr.imageName
+}
+
+// Command returns an optional command that overrides the default image
+// "CMD" or "ENTRYPOINT" for the Docker and Containerd (gRPC) drivers
+func (ctr *CRIContainer) Command() string {
+	return ctr.cmdOverride
+}
+
+//GetPodID return pod-id assosicated with container.
+func (ctr *CRIContainer) GetPodID() string {
+	return ctr.podID
+}
+
+// Type returns a driver type to identify the driver
+func (c *CRIDriver) Type() Type {
+	return CRI
+}
+
+// Info returns a string with information about the container engine/runtime details
+func (c *CRIDriver) Info() (string, error) {
+	version, err := (*c.runtimeClient).Version(c.context, &pb.VersionRequest{})
+	if err != nil {
+		return "", err
+	}
+
+	info := "CRI Client driver (Version: " + version.GetVersion() + ", API Version: " + version.GetRuntimeApiVersion() + " Runtime" + version.GetRuntimeName() + version.GetRuntimeVersion() + " )"
+
+	return info, nil
+}
+
+// Path returns the binary (or socket) path related to the runtime in use
+func (c *CRIDriver) Path() string {
+	return c.criSocketAddress
+}
+
+// Create will create a container instance matching the specific needs
+// of a driver
+func (c *CRIDriver) Create(name, image, cmdOverride string, detached bool, trace bool) (Container, error) {
+
+	if _, err := (*c.imageClient).PullImage(c.context, &pb.PullImageRequest{Image: &pb.ImageSpec{Image: image}}); err != nil {
+		return nil, err
+	}
+	if _, err := (*c.imageClient).PullImage(c.context, &pb.PullImageRequest{Image: &pb.ImageSpec{Image: defaultPodImage}}); err != nil {
+		return nil, err
+	}
+
+	// TODO: //Check if its not loaded, then load config files
```

Comment:
do you want to add a `loadConfigs()` here that uses `sync.Once`?

Reply:
Its not required now, as ``loadConfig()`` logic moved to ``NewCRIDriver()``. So just removing the comments.